scripts/build_canonical_gliner_dataset.py

```
@@ -0,0 +1,836 @@
+        input_root / "gliner2_pii_ptbr_reward_split" / "train.jsonl",
+        input_root / "gliner2_pii_ptbr_reward_split" / "validation.jsonl",
+    )
+    for path in paths:
+        for _, row in stream_jsonl(path):
+            if row is None:
+                continue
+            output = row.get("output") if isinstance(row, dict) else None
+            entities = output.get("entities") if isinstance(output, dict) else None
+            if isinstance(entities, dict) and entities:
+                return list(entities.keys())
+    raise RuntimeError("Could not infer canonical labels from gliner2_pii_ptbr_reward_split.")
```

Ruff (0.15.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)